components/IdentifyViewer.jsx

```diff
@@ -298,16 +318,31 @@ class IdentifyViewer extends React.Component {
     removeResult = (layer, result) => {
         this.setState((state) => {
             const newResultTree = {...state.resultTree};
+            const newOriginalResultTree = {...state.originalResultTree};
             newResultTree[layer] = state.resultTree[layer].filter(item => item !== result);
+            newOriginalResultTree[layer] = state.originalResultTree[layer].filter(item => item !== result);
             if (isEmpty(newResultTree[layer])) {
                 delete newResultTree[layer];
+                delete newOriginalResultTree[layer];
             }
+            const selectedLayer = isEmpty(newResultTree[layer]) ? "all" : state.selectedLayer;
             return {
                 resultTree: newResultTree,
-                currentResult: state.currentResult === result ? null : state.currentResult
+                originalResultTree: newOriginalResultTree,
+                currentResult: state.currentResult === result ? null : state.currentResult,
+                selectedLayer: selectedLayer
             };
         });
     };
+    filterResult = () => {
+        let newResultTree;
+        if (this.state.selectedLayer === "all") {
+            newResultTree = this.state.originalResultTree;
+        } else {
+            newResultTree = { [this.state.selectedLayer]: this.state.originalResultTree[this.state.selectedLayer] };
+        }
+        this.setState({resultTree: newResultTree});
```

Comment:
You should not (directly or indirectly) use `this.state` in `setState`. Rather i.e.
```
this.setState((state) => {
     return {resultTree: state.originalResultTree};
});
```
Aside from this, I think it would be simpler to just filter
```
const contents = Object.keys(this.state.resultTree).map(layer => this.renderLayer(layer));
```
according to the selected layer in `render` instead of storing a separate result tree in the state.

Reply:
Ok my solution was actually for the flast list, I wasn't really aware of that DisplaylyerTree. Then all I need is the possibility of showing all the result of the same layer maybe with a little button next to the delete one. 

I'll try something new then i'll let you know.